v0/destinations/yahoo_dsp/transform.js

```diff
@@ -49,80 +46,83 @@ const responseBuilder = async (message, destination) => {
    * The below written switch case is used to build the response for each of the supported audience type.
    *  eg. ["email", "deviceId", "ipAddress", "mailDomain", "pointOfInterest"].
    */
-  switch (audienceType) {
+  switch (AUDIENCE_TYPE[audienceType]) {
     case "email":
       // creating the output payload using the audience list and Config
-      outputPayload = createPayload(listDataKey, destination.Config);
+      dspListPayload = createPayload(traitsList, Config);
       break;
     case "deviceId":
       // throwing error if seedListType is not provided for deviceId type audience
       if (
         !seedListType ||
-        (seedListType && seedListType !== "IDFA" && seedListType !== "GPADVID")
+        (seedListType !== "IDFA" && seedListType !== "GPADVID")
       ) {
         throw new CustomError(
           `[Yahoo_DSP]:: seedListType is required for deviceId type audience and it should be any one of 'IDFA' and 'GPADVID'`,
           400
         );
       }
-      outputPayload = createPayload(listDataKey, destination.Config);
-      // updatig seedListType here
-      outputPayload = {
-        ...outputPayload,
+      dspListPayload = createPayload(traitsList, Config);
+      dspListPayload = {
+        ...dspListPayload,
         seedListType
       };
       break;
     case "ipAddress":
-      outputPayload = createPayload(listDataKey, destination.Config);
-      // updating seedListType in outputPayload
-      outputPayload = {
-        ...outputPayload,
+      dspListPayload = createPayload(traitsList, Config);
+      dspListPayload = {
+        ...dspListPayload,
         seedListType: "SHA256IP"
       };
       break;
     case "mailDomain":
       // traversing through every element in the add array for the elements to be added.
-      listDataKey.forEach(element => {
+      traitsList.forEach(userTraits => {
         // storing keys of an object inside the add array.
-        const keys = Object.keys(element);
+        const traits = Object.keys(userTraits);
         // For mailDomain the mailDomain or categoryIds must be present. throwing error if not present.
         /**
          * If mailDomain is not provided categoryIds is required. The audience includes consumers who have received
          * mail from a domain belonging to the specified categories.
          * Reference for use case of categoryIds:
          * https://developer.yahooinc.com/dsp/api/docs/traffic/audience/mrt-audience.html#:~:text=Optional-,categoryIds,-Specifies%20an%20array
          */
-        if (!(keys.includes(audienceType) || keys.includes("categoryIds"))) {
+        if (
+          !(
+            traits.includes(AUDIENCE_TYPE[audienceType]) ||
+            traits.includes("categoryIds")
```

Comment:
Please store string "categoryIds" in a constant and use that

Reply:
done